src/Analyzers.CodeFixes/CSharp/CodeFixes/OptimizeStringBuilderAppendCallCodeFixProvider.cs

```diff
@@ -258,13 +258,19 @@ public override async Task RegisterCodeFixesAsync(CodeFixContext context)
                         {
                             case 1:
                                 {
+                                    ExpressionSyntax argumentExpression = arguments[0].Expression;
+                                    if (argumentExpression is not LiteralExpressionSyntax)
+                                    {
+                                        argumentExpression = ParenthesizedExpression(argumentExpression);
+                                    }
+
                                     ArgumentListSyntax argumentList = ArgumentList(
                                         Argument(invocationInfo.Expression),
                                         arguments[0],
                                         Argument(
                                             SubtractExpression(
                                                 SimpleMemberAccessExpression(invocationInfo.Expression, IdentifierName("Length")),
-                                                arguments[0].Expression)));
+                                                argumentExpression)));
```

Comment:
Instead of ParenthesizedExpression use `arguments[0].Expression.Parenthesize()`. The parentheses will be removed if not necessary.

Reply:
Wow, there are so many great helper methods in this project! Updated the code.